lemminx-maven/src/main/java/org/eclipse/lemminx/maven/MavenHoverParticipant.java

```diff
@@ -182,19 +205,59 @@ private String collectGoal(IPositionRequest request) {
+		Set<MojoParameter> parameters;
 		try {
-			parameters = MavenPluginUtils.collectPluginConfigurationParameters(request, cache, repoSession, pluginManager);
+			parameters = MavenPluginUtils.collectPluginConfigurationMojoParameters(request, cache, repoSession, pluginManager, buildPluginManager, mavenSession);
 		} catch (PluginResolutionException | PluginDescriptorParsingException | InvalidPluginDescriptorException e) {
 			e.printStackTrace();
 			return null;
 		}
 		DOMNode node = request.getNode();
+		String parentName = node.getParentNode().getLocalName();
+		if (parentName != null && parentName.equals("configuration")) {
+			// The configuration element being hovered is at the top level
+			for (MojoParameter parameter : parameters) {
+				// TODO: Is !parameterMultiple() really required?
+				if (node.getLocalName().equals(parameter.getName()) && !parameter.isMultiple()) {
+					return MavenPluginUtils.getMarkupDescription(parameter).getValue();
+				}
+			}
+		}
 		
-		for (Parameter parameter : parameters) {
-			if (node.getLocalName().equals(parameter.getName())) {
-				return MavenPluginUtils.getMarkupDescription(parameter).getValue();
+		// Nested case: node is a grand child of configuration
+		
+		// Get the node's ancestor which is a child of configuration
+		DOMNode parentParameterNode = DOMUtils.findAncestorThatIsAChildOf(request, "configuration");
+		if (parentParameterNode != null) {
+			List<MojoParameter> parentParameters = parameters.stream()
+					.filter(mojoParameter -> mojoParameter.getName().equals(parentParameterNode.getLocalName()))
+					.collect(Collectors.toList());
+			if (!parentParameters.isEmpty()) {
+				MojoParameter parentParameter = parentParameters.get(0);
+				
+				if (parentParameter.getNestedParameters().size() == 1) {
+					// The parent parameter must be a collection of a type
+					MojoParameter nestedParameter = parentParameter.getNestedParameters().get(0);
+					Class<?> potentialInlineType = PlexusConfigHelper.getRawType(nestedParameter.getParamType());
+					if (potentialInlineType != null && PlexusConfigHelper.isInline(potentialInlineType)) {
+						return MavenPluginUtils.getMarkupDescriptionUsingParent(nestedParameter, parentParameter).getValue();
+					}
+				}
+				
+				// Get all deeply nested parameters
+				List<MojoParameter> nestedParameters = parentParameter.getFlattenedNestedParameters();
+				nestedParameters.add(parentParameter);
+				for (MojoParameter parameter : nestedParameters) {
+					if (node.getLocalName().equals(parameter.getName())) {
+						return parameter.getDescription() == null
+								? MavenPluginUtils.getMarkupDescriptionUsingParent(parameter, parentParameter)
+										.getValue()
+								: MavenPluginUtils.getMarkupDescription(parameter).getValue();
+					}
+				}
+				// Fallback case is to return parent's hover info
+				return MavenPluginUtils.getMarkupDescription(parentParameter).getValue();
```

Comment:
It’s probably best just to return null here instead, as the configuration element could be invalid